src/v/serde/test/fuzz.cc

```diff
@@ -0,0 +1,182 @@
+#include "generated_structs.h"
+#include "reflection/type_traits.h"
+
+#include <fstream>
+#include <iostream>
+
+bool operator==(iobuf const&, iobuf const&) { return true; }
+bool operator!=(iobuf const&, iobuf const&) { return true; }
+bool operator<=(iobuf const&, iobuf const&) { return true; }
+bool operator>=(iobuf const&, iobuf const&) { return true; }
+bool operator<(iobuf const&, iobuf const&) { return true; }
+bool operator>(iobuf const&, iobuf const&) { return true; }
+
+struct data_gen {
+    data_gen(std::uint8_t const* data, std::size_t const size)
+      : _data{data}
+      , _size{size} {}
+    template<
+      typename T,
+      std::enable_if_t<std::is_trivially_copyable_v<T>>* = nullptr>
+    T get() {
+        auto val = T{};
+        for (auto i = 0U; i != sizeof(val); ++i) {
+            auto const byte = get_byte();
+            std::memcpy(reinterpret_cast<std::uint8_t*>(&val) + i, &byte, 1);
+        }
+        return val;
+    }
+
+    std::uint8_t get_byte() {
+        auto const d = _data[_i];
+        ++_i;
+        if (_i == _size) {
+            _i = 0U;
+        }
+        return d;
+    }
+
+    std::uint8_t const* _data{};
+    std::size_t _size{};
+    std::size_t _i{};
+};
+
+template<typename T>
+void init(T& t, data_gen& gen) {
+    if constexpr (serde::is_envelope_v<T>) {
+        serde::envelope_for_each_field(t, [&](auto& f) { init(f, gen); });
+    } else if constexpr (reflection::is_std_optional_v<T>) {
+        if (gen.get<std::uint8_t>() > 128) {
+            t = std::make_optional<typename std::decay_t<T>::value_type>();
+            init(*t, gen);
+        } else {
+            t = std::nullopt;
+        }
+    } else if constexpr (reflection::is_std_vector_v<T>) {
+        t.resize(gen.get<uint8_t>());
+        for (auto& v : t) {
+            init(v, gen);
+        }
+    } else if constexpr (std::is_same_v<ss::sstring, std::decay_t<T>>) {
+        t.resize(gen.get<uint8_t>());
+        for (auto& v : t) {
+            v = gen.get<char>();
+        }
+    } else if constexpr (std::is_same_v<iobuf, std::decay_t<T>>) {
+        // TODO fill iobuf like string
```

Comment:
you can do something like take a ss::string constructed like above and call something like std::move(s).release() and that gives you a temporary_buffer with the contents of the string and then append the temporary_buffer to an empty iobuf